packages/cli/lib/sandboxes.js

```diff
@@ -163,16 +193,18 @@ function pollSyncTaskStatus(accountId, taskId, syncStatusUrl) {
     );
     process.exit(EXIT_CODES.SUCCESS);
   };
-  handleExit(onTerminate);
-  handleKeypress(key => {
-    if (
-      (key && key.ctrl && key.name == 'c') ||
-      key.name === 'enter' ||
-      key.name === 'return'
-    ) {
-      onTerminate();
-    }
-  });
+  if (allowEarlyTermination) {
+    handleExit(onTerminate);
+    handleKeypress(key => {
+      if (
+        (key && key.ctrl && key.name == 'c') ||
+        key.name === 'enter' ||
+        key.name === 'return'
+      ) {
+        onTerminate();
+      }
+    });
+  }
```

Comment:
Optional keypress handlers in case we want to run a sync in a different flow where there are steps required after the sync.